### PR TITLE
Desktop UI components

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Converts the rest of the components in libs/ui to either work with the desktop theme or at least use the new color/size theme fields. Most of these changes involved only minor visual differences, so I grouped them all in one PR.

Also adds spacing guidelines and generally tries to implement those for spacing consistency across components.

_Review by commit_

## Demo Video or Screenshot
Included in relevant commits

## Testing Plan
Existing automated tests plus manual visual testing

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/ui/src/themes/spacing.stories.tsx
Comment (jonahkagan) [inline libs/ui/src/themes/spacing.stories.tsx:1]: Screenshot 2023-11-02 at 10 51 29 AM
Screenshot 2023-11-02 at 10 51 42 AM

File: libs/ui/src/__snapshots__/button_bar.test.tsx.snap
Comment (jonahkagan) [inline libs/ui/src/__snapshots__/button_bar.test.tsx.snap:1]: Screenshot 2023-11-02 at 10 52 57 AM

File: libs/ui/src/modal.tsx
Comment (jonahkagan) [inline libs/ui/src/modal.tsx:1]: Screenshot 2023-11-02 at 10 55 17 AM

**Review by adghayes — APPROVED** [review]
LGTM!

We now have some components that are meant for both desktop and touch, and some that are only for touch. I don't know how much of a problem it would be in practice, but I wish there were some way we clearly distinguished which components should be used where.

**Comment by jonahkagan** [comment]
> We now have some components that are meant for both desktop and touch, and some that are only for touch.

Yeah it's a good point. I'll think about this. Let me know if you have ideas

**Comment by adghayes** [comment]
@jonahkagan I wonder if we could have a custom, required flag in our storybook types, so if we add something to storybook we have to indicate whether it's one or the other or both, and it could appear somewhere in the storybook interface.

**Comment by jonahkagan** [comment]
@adghayes something like that sounds reasonable

**Pull request merged**